judge/views/user.py

```diff
@@ -335,6 +338,59 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class WCIPEGMergeActivate(LoginRequiredMixin, TitleMixin, FormView):
+    template_name = 'user/wcipeg-merge-activate.html'
+    form_class = WCIPEGMergeActivationForm
+
+    def merge(self, from_user, to_user):
+        from_user.merge_user = to_user
+        from_user.save()
+        with transaction.atomic():
+            Submission.objects.filter(user=from_user).update(user=to_user)
+            Comment.objects.filter(user=from_user).update(user=to_user)
+            CommentVote.objects.filter(voter=from_user).update(voter=to_user)
+
+    def dispatch(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
```

Comment:
This feels a bit awkward, since you have to think "wait, what happens when the user is not authenticated?", before realizing that `LoginRequiredMixin` occurs _after_. Maybe we should use `UserPassesTestMixin` with a `test_func`. If we do, remember to place `LoginRequiredMixin` *before* `UserPassesTestMixin`.